File: heranca/java/Pessoa.java
// Disciplina: Programação Orientada a Objetos
// Professor : Gilson Pereira
// Assunto   : Herança em Java
// Fonte     : Ascencio e Campos - Fundamentos da Programação de Computadores (3ª ed.)

public class Pessoa {
	private String nome;
	private String endereco;
	private String dataNascimento;

	public Pessoa() {
		System.out.println("Executando construtor da classe Pessoa");
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getEndereco() {
		return endereco;
	}

	public void setEndereco(String endereco) {
		this.endereco = endereco;
	}

	public String getDataNascimento() {
		return dataNascimento;
	}

	public void setDataNascimento(String dataNascimento) {
		this.dataNascimento = dataNascimento;
	}
}

File: heranca/cpp/Pessoa.cpp
// Disciplina: Programação Orientada a Objetos
// Professor : Gilson Pereira
// Assunto   : Herança em C++
// Fonte     : Ascencio e Campos - Fundamentos da Programação de Computadores (3ª ed.)

#include "Pessoa.hpp"
#include <iostream>

using namespace std;

Pessoa::Pessoa()
{
	cout << "\nExecutando construtor da classe Pessoa";
}

string Pessoa::getNome()
{
    return nome;
}

void Pessoa::setNome(string n)
{
    nome = n;
}

string Pessoa::getEndereco()
{
    return endereco;
}

void Pessoa::setEndereco(string e)
{
    endereco = e;
}

string Pessoa::getDataNascimento()
{
    return dataNascimento;
}

void Pessoa::setDataNascimento(string d)
{
    dataNascimento = d;
}